coriolis/tests/osmorphing/osmount/test_base.py

```diff
@@ -0,0 +1,1014 @@
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_get_mounted_devices')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_find_and_mount_root')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_find_dev_with_contents')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_get_volume_block_devices')
+    @mock.patch.object(
+        base.BaseLinuxOSMountTools, '_check_mount_fstab_partitions'
+    )
+    def test_mount_os_run_xfs(self, mock_check_mount_fstab_partitions,
+                              mock_get_volume_block_devices,
+                              mock_find_dev_with_contents,
+                              mock_find_and_mount_root,
+                              mock_get_mounted_devices, mock_get_vgs,
+                              mock_exec_cmd):
+        mock_get_volume_block_devices.return_value = ["/dev/sda", "/dev/sdb"]
+        mock_find_and_mount_root.return_value = ("/tmp/tmp_dir", "/dev/sda")
+        mock_get_vgs.return_value = {"vg1": "/dev/sda"}
+        mock_exec_cmd.return_value = b"xfs\n"
+
+        result = self.base_os_mount_tools.mount_os()
+
+        mock_get_volume_block_devices.assert_called_once()
+        mock_find_dev_with_contents.assert_called_once_with(
+            ['xfs', 'xfs'], one_of_files=["grub", "grub2"])
+        mock_get_mounted_devices.assert_called_once()
+        mock_find_and_mount_root.assert_called_once()
+        mock_get_vgs.assert_called_once()
+        mock_check_mount_fstab_partitions.assert_called_once_with(
+            "/tmp/tmp_dir", mountable_lvm_devs=['xfs'])
+
+        mock_exec_cmd.assert_has_calls([
+            mock.call("sudo partx -v -a /dev/sda || true"),
+            mock.call("sudo ls -1 /dev/sda*"),
+            mock.call("sudo partx -v -a /dev/sdb || true"),
+            mock.call("sudo ls -1 /dev/sdb*"),
+            mock.call("sudo vgck"),
+            mock.call("readlink -en xfs"),
+            mock.call('sudo blkid -o value -s TYPE xfs || true'),
+            mock.call("readlink -en xfs"),
+            mock.call('sudo blkid -o value -s TYPE xfs || true'),
+            mock.call('sudo mount %s "/tmp/tmp_dir/boot"' %
+                      mock_find_dev_with_contents.return_value),
+            mock.call("sudo lvdisplay -c"),
+        ])
+
+        self.assertEqual(result, ('/tmp/tmp_dir', '/dev/sda'))
+
+    @mock.patch.object(base.BaseSSHOSMountTools, '_exec_cmd')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_get_vgs')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_get_mounted_devices')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_find_and_mount_root')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_find_dev_with_contents')
+    @mock.patch.object(base.BaseLinuxOSMountTools, '_get_volume_block_devices')
+    @mock.patch.object(
+        base.BaseLinuxOSMountTools, '_check_mount_fstab_partitions'
+    )
+    def test_mount_os_check_fs(self, mock_check_mount_fstab_partitions,
```

Comment:
I believe that this test case is not needed since the suggested `test_mount_os` above uses a valid file system, and the code with `utils.check_fs` is touched.

However, in the suggested `test_mount_os` above, please add validation so `utils.check_fs` call is asserted.